presto-tests/src/test/java/com/facebook/presto/tests/TestQueryManager.java

```diff
@@ -226,4 +227,37 @@ public void testQueryOutputSizeExceeded()
             assertEquals(queryInfo.getErrorCode(), EXCEEDED_OUTPUT_SIZE_LIMIT.toErrorCode());
         }
     }
+
+    @Test
+    public void testQueuedQueryCount()
+            throws Exception
+    {
+        DispatchManager dispatchManager = queryRunner.getCoordinator().getDispatchManager();
+        // Create 5 queries to guarantee queueing
+        createQueries(dispatchManager, 5);
+
+        assertNotEquals(dispatchManager.getStats().getQueuedQueries(), 0L);
+
+        assertEquals(
```

Comment:
This test only tests the stats at one snapshot. Can we enhance it to multiple checks before all queries finish such that all the states match? Something like
```
while (stats.queuedQueries + stats.runningQueries != 0) {
    assert queuedQueries match;
    assert runningQueries match;
    sleep(x);
}
```